src/atoms/affine/dot.jl

```diff
@@ -1,26 +1,19 @@
 import LinearAlgebra.dot
-export vecdot, dot
+export dot
 
+ismatrix(x::AbstractExpr) = (s = size(x); length(s) == 2 && !any(==(1), s))
```

Comment:
Perhaps,

```suggestion
function ismatrix(x::AbstractExpr)
    s = size(x)
    return length(s) == 2 && s[1] != 1 && s[2] != 1
```

The `(a; b)` form, I find hard to read.
And using `any` on something of length 2 seems both harder to read and slow.